app/services/journeyAnswers/IncomeAnswersService.scala

```diff
@@ -80,13 +81,21 @@ class IncomeAnswersServiceImpl @Inject() (repository: JourneyAnswersRepository,
       _        <- upsertPeriodSummary(response, ctx, answers)
       _        <- maybeUpsertData.traverse(upsertData => EitherT(connector.createAmendSEAnnualSubmission(upsertData))).leftAs[ServiceError]
       _        <- repository.upsertAnswers(JourneyContext(taxYear, businessId, mtditid, Income), Json.toJson(storageAnswers))
+      _        <- maybeDeleteExpenses(ctx, answers)
     } yield ()
 
     result.leftAs[ServiceError]
   }
 
+  private def maybeDeleteExpenses(ctx: JourneyContextWithNino, answers: IncomeJourneyAnswers): EitherT[Future, ServiceError, Unit] =
+    if (answers.tradingAllowance == UseTradingAllowance) {
```

Comment:
maybe we can change it to pattern match, then when we add a new type the code won't compile because of the 'sealed'?

answers.tradingAllowace match {
  case UstTradingAllowance => repository.delete...
   case DeclareExpenses => EitherT.rightT
}